src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1912,6 +1912,11 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
     /// Before actually creating the table, check if it will lead to cyclic dependencies.
     checkTableCanBeAddedWithNoCyclicDependencies(create, query_ptr, getContext());
 
+    bool is_initial_query = getContext()->getClientInfo().query_kind == ClientInfo::QueryKind::INITIAL_QUERY ||
+                            (getContext()->getZooKeeperMetadataTransaction() && !getContext()->getZooKeeperMetadataTransaction()->isInitialQuery());
+    if (!internal && is_initial_query)
+        throwIfTooManyEntities(create, res);
```

Comment:
One small detail I forgot about in my previous PR is that Initial queries in Database Replicated have  `query_kind == ClientInfo::QueryKind::SECONDARY_QUERY`  at this point in the code, and we need to check for getZooKeeperMetadataTransaction() to find it out.

The second thing is that this check does not require storage being created so it is better to perform this check before the creation of a bunch of unaccounted ZooKeeper nodes in case of RMT tables.